pkg/sqlutil/pg/connection.go

```diff
@@ -0,0 +1,121 @@
+package pg
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/XSAM/otelsql"
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/stdlib"
+	"github.com/jmoiron/sqlx"
+	"github.com/scylladb/go-reflectx"
+	"go.opentelemetry.io/otel"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+
+	// need to make sure pgx driver is registered before opening connection
+	_ "github.com/jackc/pgx/v4/stdlib"
+)
+
+// NOTE: This is the default level in Postgres anyway, we just make it explicit here.
+const defaultIsolation = sql.LevelReadCommitted
+
+// Driver is a compiler enforced type used that maps to database driver names
+type Driver string
+
+const (
+	// DriverPostgres represents the postgres dialect.
+	DriverPostgres Driver = "pgx"
+	// DriverTxWrappedPostgres is useful for tests.
+	// When the connection is opened, it starts a transaction and all
+	// operations performed on the DB will be within that transaction.
+	DriverTxWrappedPostgres Driver = "txdb"
+)
```

Comment:
I notice some overlap here with this PR:
https://github.com/smartcontractkit/chainlink-common/pull/910

I assume we only want the dialects defined in one place--so we'll need to decide whether it's here in `pg/connection.go` or in `pg/dialects.go`:
https://github.com/smartcontractkit/chainlink-common/pull/910/files#diff-57a42a4e715a6082c3fd24971c8516609094df5e15be2ae9fe49f302a2cc3efa

I'm fine with either, just pointing out that this is going to cause conflicts.

Reply:
Why don't we combine packages?

Reply:
I had tried to ignore txdb in this PR, but the switch for the different API was required :shrug:

Reply:
Oh, yeah I hadn't noticed this but your is `pkg/sqlutil/pg` and mine is `pkg/pg`... I don't have a strong preference so if you feel it should be the first one I can move mine.

Reply:
I think it would be good to nest the packages but I don't have strong preferences on type names